### PR TITLE
Refactor mason update

### LINKED ISSUE #13211: Mason Update Interface
Currently the interface for Mason Update needs to be changed in order to support passing arguments into the file. Mason Update does not have a help output or any arguments currently. this was primarily because a user hardly ever just updates a toml file without building, so the function is mostly used internally in mason. However, as more users are coming to mason, little things like this need to be fleshed out.

###  Current Interface
```chapel
proc UpdateLock(args: [] string, tf="Mason.toml", lf="Mason.lock") 
```

### Suggested Interface
```chapel
proc MasonUpdate(args: [] string)
```

The UpdateLock function will still not be changed, and the new interface will not be called internally. The new interface will only be called in the event that a user calls ```mason update <arg>```. No existing tests will need to be modified with this change, but some tests should be added. 

this issue came up in the development of #13199 @ben-albrecht @oplambeck

### PR DESCRIPTION
**Pull request by ankingcodes**
fixes #13211 & #14990
- [x] Fix bug where `mason update --help` doesn't show help message and runs update instead.
- [x] Refactor mason update function.
- [x] Select style arg parsing

### CODE REVIEW

**Comment by ankingcodes** [comment]
@ben-albrecht please add a GSoC label here !

**Comment by ben-albrecht** [comment]
> Is select style arg parsing really required here?

While it may not be useful in `mason update`, I'm OK with continuing to use it for consistency across all mason interface implementations.

File: tools/mason/MasonUpdate.chpl [hunk 1]
```diff
@@ -45,13 +46,33 @@ The current resolution strategy for Mason 0.1.0 is the IVRS as described below:
 */
 
 private var failedChapelVersion: list(string);
-
+private var update = false;
 //
 // Temporary passthrough transforming array to list to appease the compiler.
 //
-proc UpdateLock(args: [?d] string, tf="Mason.toml", lf="Mason.lock") {
```
Comment (ben-albrecht) [inline tools/mason/MasonUpdate.chpl:52]: We don't use these args anywhere else anymore?

Comment (ankingcodes) [reply tools/mason/MasonUpdate.chpl:52]: I believe only `mason update` from `mason.chpl` called this function. Other than that, I didn't get any error else where

Comment (ben-albrecht) [reply tools/mason/MasonUpdate.chpl:52]: OK thanks!

**Pull request merged**